android_backup/android_backup.py

```diff
@@ -165,19 +175,11 @@ def unpack(self):
         with open(pickle_fname, 'wb') as fp:
             pickle.dump(members, fp)
 
-    def list(self):
-        data = self.fp.read()
-
-        if self.encryption == EncryptionType.AES256:
-            data = self._decrypt(data)
-
-        if self.compression == CompressionType.ZLIB:
-            data = zlib.decompress(data, zlib.MAX_WBITS)
-
-        tar = tarfile.TarFile(fileobj=io.BytesIO(data))
-        tar.list()
+    def list(self, password=None):
+        tar = self.read_tar(password)
```

Comment:
I guess this should be ``read_data`` instead of ``read_tar``?

Reply:
Indeed, I did not use that at all so missed it, my bad!